languages/multi-language/src/main/java/de/jplag/multilang/MultiLanguageParser.java

```diff
@@ -0,0 +1,36 @@
+package de.jplag.multilang;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import de.jplag.Language;
+import de.jplag.ParsingException;
+import de.jplag.Token;
+
+public class MultiLanguageParser {
+    private final List<Language> languages;
+
+    public MultiLanguageParser(MultiLanguageOptions options) {
+        this.languages = options.getLanguages();
+    }
+
+    public List<Token> parseFiles(Set<File> files, boolean normalize) throws ParsingException {
+        List<Token> results = new ArrayList<>();
+        for (File file : files) {
+            Optional<Language> language = findLanguageForFile(file);
+            if (language.isPresent()) {
+                results.addAll(language.get().parse(Set.of(file), normalize));
+            }
+        }
+        return results;
+    }
+
+    private Optional<Language> findLanguageForFile(File file) {
+        return this.languages.stream().filter(language -> Arrays.stream(language.suffixes()).anyMatch(suffix -> file.getName().endsWith(suffix)))
+                .findFirst();
```

Comment:
This will currently pick the first language for cases where multiple languages support the same file type. Have you discussed this behavior? This affects the C/C++ modules and also the EMF modules.

Reply:
(also in future Java vs. Java-CPG)

Reply:
Ah I understand; I overlooked that the users specify the languages when using the module.
In the long run, I considered making the language module the default, but that would only be user-friendly if users do not need to specify languages. This would mean the multi-language module automatically parses all code that JPlag supports. Then we need prioritization.

With the current solution, we add yet another cli argument, which is less likely to be used by many users. For now, let us leave it as is, but before the release, we need to think about which mode we truly want. If we want more people to try out the language module, we probably need to implement the unparameterized version. However, in all cases, I would not make it the default language straight away.